### PR TITLE
Fix zfits boardtime

### PR DESCRIPTION
**Pull request by MaxNoe**

### CODE REVIEW

File: src/main/java/fact/io/hdureader/ZFITSHeapReader.java [hunk 1]
```diff
@@ -674,43 +674,52 @@ private Serializable readSingleValueFromBuffer(BinTable.TableColumn c, ByteBuffe
...
+            case CHAR:
+                char[] chars = new char[c.repeatCount];
+                for (int i = 0; i < c.repeatCount; i++) {
+                    chars[i] = buffer.getChar();
+                }
+                return chars;
             case SHORT:
                 short[] shorts = new short[c.repeatCount];
-                buffer.asShortBuffer().get(shorts);
+                for (int i = 0; i < c.repeatCount; i++) {
```
Comment (mackaiver) [inline src/main/java/fact/io/hdureader/ZFITSHeapReader.java]: does this have a performance impact?

**Comment by MaxNoe** [comment]
What's the Performance impact if it was wrong before?

File: src/main/java/fact/io/hdureader/ZFITSHeapReader.java (hunk 1 above)
Comment (MaxNoe) [reply src/main/java/fact/io/hdureader/ZFITSHeapReader.java]: What's the Performance impact if it was wrong before?

**Comment by dneise** [comment]
Is the fact that the buffer was not advanced a bug in get() ?

**Comment by MaxNoe** [comment]
No, as documented in #389, the calls to `asXXXXBuffer` create a new view, that by design has an independent position. So this is the intended behaviour. We just used it wrong.

**Comment by dneise** [comment]
Ah I missed that link to the documentation .. so this was wrong all the time and in all the places you fixed it ... and it was never realized. ... that is ... somehow depressing.

File: src/main/java/fact/io/hdureader/ZFITSHeapReader.java (hunk 1 above)
Comment (mackaiver) [reply src/main/java/fact/io/hdureader/ZFITSHeapReader.java]: About a factor of 6 if I remember correctly. Also to my understanding the `buffer.asShortBuffer().get(shorts);` should be equivalent to the loop. I don't understand where this goes wrong.

**Comment by mackaiver** [comment]
> No, as documented in #389, the calls to `asXXXXBuffer` create a new view, that by design has an independent position. So this is the intended behaviour. We just used it wrong.

Aha. That's surprising.

**Comment by mackaiver** [comment]
Can we advance the buffer by hand? We replaced all those loops a while ago on purpose.

**Comment by MaxNoe** [comment]
Replaced the loops with reading into arrays by directly putting the correct buffer views into the tile cache.

**Pull request merged**